### PR TITLE
Add support for Ubuntu 20.04

### PR DESCRIPTION
**Pull request by ddeidda**
This commit adds the changes to parameters, validators and cloudformation templates needed to support the new base_os Ubuntu 20.04.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,9 +5,9 @@ CHANGELOG
 ------
 **ENHANCEMENTS**
 
+- Add support for Ubuntu 20.04
 - Add validation to prevent using a `cluster_resource_bucket` that is in a different region than the cluster.
-- Add validation for `cluster_type` configuration parameter in `cluster` section
-- Add validation for `compute_type` configuration parameter in `queue` section
+
```
Comment (rexcsn) [inline CHANGELOG.md:10]: nit: an extra line?

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,7 +26,9 @@ CHANGELOG
 ------
 **ENHANCEMENTS**
 
-- Enable support for ARM instances in China and GovCloud regions when using Ubuntu 18.04 or Amazon Linux 2. 
+- Enable support for ARM instances in China and GovCloud regions when using Ubuntu 18.04 or Amazon Linux 2.
+- Add validation for `cluster_type` configuration parameter in `cluster` section
```
Comment (rexcsn) [inline CHANGELOG.md:30]: We might want to backport these changelog changes to 2.10 branch

Comment (ddeidda) [reply CHANGELOG.md:30]: Looks like they're already there: https://github.com/aws/aws-parallelcluster/blob/release-2.10/CHANGELOG.md.
We probably forgot to update the develop version.

File: tests/integration-tests/configs/common.jinja2 [hunk 3]
```diff
@@ -5,13 +5,13 @@
...
-{%- set OSS_CHINA_ARM = ["alinux2", "ubuntu1804"] -%}
-{%- set OSS_GOVCLOUD_ARM = ["alinux2", "ubuntu1804"] -%}
-{%- set OSS_ONE_PER_DISTRO = ["centos7", "alinux2", "ubuntu1804"] -%}
+{%- set OSS_COMMERCIAL_X86 = ["alinux2", "centos7", "centos8", "ubuntu1804", "ubuntu2004"] -%}
+{%- set OSS_CHINA_X86 = ["alinux2", "ubuntu1804", "ubuntu2004"] -%}
+{%- set OSS_GOVCLOUD_X86 = ["alinux2", "ubuntu1804", "ubuntu2004"] -%}
+{%- set OSS_COMMERCIAL_ARM = ["alinux2", "centos8", "ubuntu1804", "ubuntu2004"] -%}
+{%- set OSS_CHINA_ARM = ["alinux2", "ubuntu1804", "ubuntu2004"] -%}
+{%- set OSS_GOVCLOUD_ARM = ["alinux2", "ubuntu1804", "ubuntu2004"] -%}
+{%- set OSS_ONE_PER_DISTRO = ["centos7", "alinux2", "ubuntu1804", "ubuntu2004"] -%}
```
Comment (rexcsn) [inline tests/integration-tests/configs/common.jinja2:14]: Should we only keep 1 ubuntu OS here?

File: tests/integration-tests/configs/common/common.yaml [hunk 4]
```diff
@@ -168,12 +168,12 @@ disable_hyperthreading:
...
       - regions: ["us-west-1"]
         instances: ["c5.xlarge"]
-        oss: ["ubuntu1804"]
+        oss: ["ubuntu1804", "ubuntu2004"]
         schedulers: ["slurm"]
   test_disable_hyperthreading.py::test_sit_disable_hyperthreading:
     dimensions:
       # Manually disabled HT
-      {%- for os, scheduler in [("centos7", "torque"), ("ubuntu1804", "sge")] %}
+      {%- for os, scheduler in [("ubuntu2004", "torque"), ("ubuntu1804", "sge")] %}
```
Comment (rexcsn) [inline tests/integration-tests/configs/common/common.yaml:176]: Is this change from centos7 to ubuntu20 intended?

File: tests/integration-tests/configs/common.jinja2 (hunk 3 above)
Comment (ddeidda) [reply tests/integration-tests/configs/common.jinja2:14]: Absolutely yes! I left Ubuntu1804; let's talk if you think we should replace it with Ubuntu2004

File: tests/integration-tests/configs/common/common.yaml (hunk 4 above)
Comment (ddeidda) [reply tests/integration-tests/configs/common/common.yaml:176]: Yes, the reasoning behind being that centos7 with manually disabled ht is already tested in `test_hit_disable_hyperthreading`. So this way both `centos7` and `ubuntu20` will be tested in both use cases. What do you think?

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,9 +5,8 @@ CHANGELOG
 ------
 **ENHANCEMENTS**
 
+- Add support for Ubuntu 20.04
```
Comment (enrico-usai) [inline CHANGELOG.md:8]: minor: we are using a final dot.

File: CHANGELOG.md [hunk 6]
```diff
@@ -26,7 +25,9 @@ CHANGELOG
 ------
 **ENHANCEMENTS**
 
-- Enable support for ARM instances in China and GovCloud regions when using Ubuntu 18.04 or Amazon Linux 2. 
+- Enable support for ARM instances in China and GovCloud regions when using Ubuntu 18.04 or Amazon Linux 2.
+- Add validation for `cluster_type` configuration parameter in `cluster` section
+- Add validation for `compute_type` configuration parameter in `queue` section
```
Comment (enrico-usai) [inline CHANGELOG.md:30]: same here: final dot.

File: tests/integration-tests/configs/common.jinja2 (hunk 3 above)
Comment (enrico-usai) [reply tests/integration-tests/configs/common.jinja2]: I think the name of the variable highlight we should have a single os per distro, I'd remove ubuntu1804

File: CHANGELOG.md (hunk 6 above)
Comment (ddeidda) [reply CHANGELOG.md]: Fixed

File: CHANGELOG.md (hunk 5 above)
Comment (ddeidda) [reply CHANGELOG.md]: Fixed

File: tests/integration-tests/configs/common.jinja2 (hunk 3 above)
Comment (ddeidda) [reply tests/integration-tests/configs/common.jinja2]: I removed Ubuntu2004 to avoid disrupting the existing set; let's talk if we prefer Ubuntu20

**Review by enrico-usai — APPROVED** [review]
LGTM!

**Pull request merged**